### PR TITLE
RTP_175 / Added support for NEXT_DAY Function

### PR DESCRIPTION
**Pull request by rushikeshsapate-dm**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -11294,4 +11294,49 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
...
+  @Test public void testNextDayFunctionWithDate() {
+    final RelBuilder builder = relBuilder();
+    final RexNode nextDayRexNode = builder.call(SqlLibraryOperators.NEXT_DAY,
+        builder.literal("2023-02-22"), builder.literal("TUESDAY"));
+    final RelNode root = builder
+        .scan("EMP")
+        .project(builder.alias(nextDayRexNode, "next_day"))
+        .build();
+    final String expectedSpark =
+        "SELECT NEXT_DAY('2023-02-22', 'TUESDAY') next_day\n"
```
Comment (vijayjogi-dm) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11307]: Plz use existing constant or create new for the day (TUESDAY). Also add one more test case with other day.

Comment (rushikeshsapate-dm) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11307]: Done

**Review by vijayjogi-dm — COMMENTED** [review]
L1 review is done

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 2]
```diff
@@ -1417,4 +1417,15 @@ private SqlLibraryOperators() {
...
+  @LibraryOperator(libraries = {ORACLE, HIVE, SPARK})
+  public static final SqlFunction NEXT_DAY =
+      new SqlFunction(
+          "NEXT_DAY",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.DATE,
+          null,
+          OperandTypes.family(SqlTypeFamily.ANY,
+              SqlTypeFamily.STRING),
+          SqlFunctionCategory.SYSTEM);
```
[the remark this answers is not in the document]
Comment (rushikeshsapate-dm) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java]: SqlFunctionCategory.TIMEDATE found to be most appropriate category so changed  SqlFunctionCategory.SYSTEM to SqlFunctionCategory.TIMEDATE

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 3]
```diff
@@ -11294,4 +11291,139 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
...
+  @Test public void testNextDayFunctionWithMonday() {
+    final RelBuilder builder = relBuilder();
+    final RexNode nextDayRexNode = builder.call(SqlLibraryOperators.NEXT_DAY,
+        builder.literal("current_date"), builder.literal(Day.MONDAY.getValue()));
+    final RelNode root = builder
+        .scan("EMP")
+        .project(builder.alias(nextDayRexNode, "next_day"))
+        .build();
+    final String expectedSpark =
+        "SELECT NEXT_DAY('current_date', 'MONDAY') next_day\n"
```
[the remark this answers is not in the document]
Comment (rushikeshsapate-dm) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: DONE -- CURRENT_DAY is called as function

File: core/src/main/java/org/apache/calcite/util/Day.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.calcite.util;
+
+public enum Day {
```
[the remark this answers is not in the document]
Comment (rushikeshsapate-dm) [reply core/src/main/java/org/apache/calcite/util/Day.java]: DONE with the help of DayOfWeek.java

**Review by deepakkadam-dm — APPROVED** [review]
Looks Ok

**Pull request merged**